### PR TITLE
Feature : conference between selective parties (#63) - version 1

### PR DESCRIPTION
**Pull request by ploufs**

### CODE REVIEW

File: tinyphone/server.cpp [hunk 1]
```diff
@@ -472,6 +472,36 @@ void TinyPhoneHttpServer::Start() {
...
+		else {
+			phone.AttendedTransfer(call, dest_call);
+			json response = {
+				{ "message",  "Attended transfer Triggered" },
+				{ "call_id" , call_id },
+				{ "dest_call_id" , dest_call_id },
+			};
+			return tp::response(200, response);
+		}
+			});
```
Comment (kingster) [inline tinyphone/server.cpp]: Please fix this indentation error

File: tinyphone/server.cpp [hunk 2]
```diff
@@ -472,6 +472,36 @@ void TinyPhoneHttpServer::Start() {
 		}
 	});
 
+	CROW_ROUTE(app, "/calls/<int>/attendedtransfer/<int>")
```
Comment (kingster) [inline tinyphone/server.cpp]: `attended-transfer` would be better instead of a single word?

File: tinyphone/phone.cpp [hunk 3]
```diff
@@ -201,6 +201,17 @@ namespace tp {
 		call->answer(prm);
 	}
 
+	void TinyPhone::AttendedTransfer(SIPCall* call, SIPCall* dest_call) {
+		CallOpParam prm;
+		prm.statusCode = pjsip_status_code::PJSIP_SC_OK;
+
+		//const Call dest = (const Call)dest_call;
+		//const Call dest = dynamic_cast<const Call>(dest_call);
+		const Call dest = *dest_call;
```
Comment (kingster) [inline tinyphone/phone.cpp]: Is this required? 

Can't we just use this instead?

```
call->xferReplaces(*dest_call,prm);
```

File: tinyphone/server.cpp [hunk 4]
```diff
@@ -472,6 +472,36 @@ void TinyPhoneHttpServer::Start() {
...
+		([&phone](int call_id,int dest_call_id) {
+		pj_thread_auto_register();
+
+		SIPCall* call = phone.CallById(call_id);
+		SIPCall* dest_call = phone.CallById(dest_call_id);
+		if (call == nullptr) {
+			return tp::response(400, {
+				{ "message", "Call Not Found" },
+				{"call_id" , call_id}
+				});
```
Comment (kingster) [inline tinyphone/server.cpp]: Indentation issue.

**Review by kingster — CHANGES_REQUESTED** [review]
Thanks for the PR

Could you have a look at the comments? Also please fix the indentation error in the new function that was added.

**Comment by ploufs** [comment]
The code is updated

**Comment by kingster** [comment]
Thank you @ploufs !

**Pull request merged**